### PR TITLE
refactor(controller): restructure controlplane commands

### PR DESCRIPTION
**Pull request by hiddeco**
This is the "control plane" version of #1547.

While it does move things into a better direction by creating a better separation of concerns, this continues to be a candidate to be revised again when we have time.

### CODE REVIEW

**Comment by krancour** [comment]
At a glance this looks fantastic, and I agree it's a step in the right direction with room for further improvement.

I do want to put this off to v0.6.0 though.

File: cmd/controlplane/controller.go [hunk 1]
```diff
@@ -32,305 +33,367 @@ import (
...
+	ShardName  string
+	KubeConfig string
+
+	ArgoCDEnabled       bool
+	ArgoCDKubeConfig    string
+	ArgoCDNamespaceOnly bool
+
+	RolloutsEnabled      bool
+	RolloutsKubeConfig   string
+	AnalysisRunNamespace string
```
Comment (krancour) [inline cmd/controlplane/controller.go:46]: This option is gone since #1706

File: cmd/controlplane/controller.go [hunk 2]
```diff
@@ -32,305 +33,367 @@ import (
...
+		// control plane cluster) and project namespaces do not exist in the
+		// shard clusters. We need a place to put them, so for now we allow
+		// the user to specify a namespace that that exists on each shard for
+		// this purpose. Note that the namespace does not need to be the same
+		// on every shard. This may be one of the weaker points in our tenancy
+		// model and can stand to be improved.
+		cacheOpts.DefaultNamespaces = map[string]cache.Config{
+			o.AnalysisRunNamespace: {},
+		}
+	}
```
Comment (krancour) [inline cmd/controlplane/controller.go:302]: This went away with #1706

File: cmd/controlplane/controller.go (hunk 1 above)
Comment (krancour) [inline cmd/controlplane/controller.go:46]: Both of these went away in #1706.

AnalysisRuns now live exclusively in the Kargo control plane, so there's never a need for a separate client nor a need to fall back on using any namespace that isn't a Project's own namespace.

File: cmd/controlplane/controller.go [hunk 3]
```diff
@@ -32,305 +33,367 @@ import (
...
+			Scheme: scheme,
+			Metrics: server.Options{
+				BindAddress: "0",
+			},
+			Cache: cacheOpts,
+		},
+	)
+}
+
+func (o *controllerOptions) setupRolloutsManager(ctx context.Context) (manager.Manager, error) {
```
Comment (krancour) [inline cmd/controlplane/controller.go:250]: Not needed since #1706

**Comment by krancour** [comment]
Apart from conflicts with #1706, this looks good.

**Comment by hiddeco** [comment]
Think I managed to rebase this correctly, but a careful look with a second pair of 👀 would be much appreciated.

**Comment by krancour** [comment]
@hiddeco it looked good before rebase... I'm just going to run it.

**Comment by krancour** [comment]
Works flawlessly. Startup logs look just like they did before. A few e2e cases all check out.

**Review by krancour — APPROVED** [review]
🔥

**Pull request merged**